firmware/application/app_settings.cpp

```diff
@@ -145,15 +146,21 @@ bool load_settings(std::string_view store_name, SettingBindings& bindings) {
             });
 
         // If found, parse the value.
-        if (it != bindings.end())
+        if (it != bindings.end()) {
             it->parse(cols[1]);
+            settings_loaded = true;
+        }
     }
 
-    return true;
+    return settings_loaded;
 }
 
 bool save_settings(std::string_view store_name, const SettingBindings& bindings) {
     File f;
+
+    if (!portapack::persistent_memory::save_app_settings())
```

Comment:
I think this breaks apps that save settings other than radio settings (like scanner).

Reply:
This check is already happening in the save method of the settingsManager.

Reply:
Maybe this should skip creating the file if the bindings are empty.

Reply:
Added check for bindings empty() per suggestion above.

I still didn't see the portapack::persistent_memory::save_app_settings() being checked elsewhere when saving settings; testing shows that an empty file was indeed being written.

IMO, it's OK if some settings are not preserved when "Save App Settings" is disabled, and actually prefer if .ini files on my SD card don't get overwritten when Save is disabled.  As long as apps still work when an .ini file is not found, then everything should still work OK if we skip writing one, with the exception of course that some settings won't get preserved.

Please "unresolve" if I haven't addressed your concerns.

Reply:
I did just think of another bug here though, if Load is turned off, but Save is turned on, it won’t actually save because there won’t be any radio bindings set up. It’s somewhat minor though because it would be kind of odd to save without wanting to load.